tests/BUILD

```diff
@@ -70,10 +70,21 @@ cc_test(
     ],
 )
 
+cc_library(
```

Comment:
This test is a cc_library and the other one is a cc_test is it supposed to be like this ?

Reply:
Yes, it is supposed to be like this. Note that here cc_library is a library that is used in three different tests.

Reply:
Ok, just to confirm you have a cc_library that has the name simulator_test.h and cc_tests that also have the _test part of their name. You don’t find that confusing ?

Reply:
I don't think this structure is completely unheard of in Google-internal code, although usually the test library is named something like "_testfixture.h" to distinguish it from the "actual" test files.

Reply:
I added `fixture` to library and file names.